common/ffsparser.h

```diff
@@ -57,6 +57,11 @@ typedef struct PROTECTED_RANGE_ {
     UINT8      Type;
     UINT8      : 8;
     UByteArray Hash;
+
+    // Additional offsets and sizes.
```

Comment:
This is not a pretty extension of the struct, but it prevented a lot of other refactoring, because all the other code assumes  `PROTECTED_RANGE` has only one offset and size.

If you have an alternate suggestion for `PROTECTED_RANGE`, I can try to incorporate it and refactor the rest of the code.

Reply:
Yeah, this is definitely somewhat dirty, let's see if we can invent a better solution. Mind sharing a test image?

Reply:
Sorry for losing track of this. The bump was very welcome.

Sample image can be downloaded from: https://www.supermicro.com/en/support/resources/downloadcenter/firmware/MBD-X12SDV-10C-SP6F/BIOS